### ISSUE TITLE
Meta Issue - Goals

### ISSUE THREAD

**Issue by pmig** [mkbouaziz/OrientR#1]
Hi @mkbouaziz ,

I see that you forked https://github.com/retrography/OrientR repos and included new features!
I am currently in a R project and we will use Orient so I have a high interest in helping this package out. Do you know why your PR's haven't been accepted as they look very promising and feature documentation?

Do you have any plans for this package in future?
What is missing?

Best, Philip

**Comment by mkbouaziz** [comment]
Hi @pmig , Welcome,

About PR haven't been accepted:
sincerely I don't know why, I contacted @retrography for this issue, but no answer yet.

I haven't any plan now, but I can work more in this package by weekends 

About file orientSql.R:
I saw that I couldn't use: Insert, update, delete in the old package of OrientR, so I decide to resolve the problem by checking the type of query. until now I haven't resolved this problem.

I added other functions for more details about Orient Databases

Best, Mohamed

**Comment by pmig** [comment]
The Orient REST API provides different interfaces:
-  [Document Interface](http://orientdb.com/docs/2.1/OrientDB-REST.html#document) which provides GET / PUT / POST Methods to update single nodes
  - currently only [get document by class](http://orientdb.com/docs/2.1/OrientDB-REST.html#document-by-class) is support
  - I would like to implement also the true document api with GET HEAD/ POST / PUT / DELETE / PATCH methods, for more information see #5
- [Query Interface](http://orientdb.com/docs/2.1/OrientDB-REST.html#query) which only performs queries that don't change the state of the orient
  
  > And can only perform idempotent commands
- [Command interface](http://orientdb.com/docs/2.1/OrientDB-REST.html#command) which can perform different update types

The Query and Command Interface in this package is strangely mixed into  the orientSql function. Whats the plan on that one?
I would recommend to split the query and execute command function again to follow the API interfaces and remove complexity.

**Comment by mkbouaziz** [comment]
about the documentation, I used the official WebSite Documentations.

there are two methods:
1- we implement one function that can execute all types of SQL query, so users will not have a problem. I implemented functions by this way.
2- we implement functions by type of query.

So our first goal is to implement successfully all REST API, then we can choose the best one.

If you want to split query I have no problem, but I think it will not be great, we can try it no problem.

**Comment by pmig** [comment]
ah ok cool. Yeah I think implementing all REST API is a good first step. Then we can fore sure choose / optimize (-:

I'm currently on the document api as described in #5

**Comment by retrography** [comment]
Hi guys,

I am happy you are getting active on this. I developed the original package just because I needed to fetch some data into R for my research and never thought it may attract further attention.  Feel free to fork and modify and put the things in order. 

I am awfully busy writing up my doctoral thesis, what explains the lack of responsiveness on my side. I do look at the activity here and I gladly accept the pull requests, but sometimes it may take me days to come back to you. 

@mkbouaziz you closed your pull request before I get to write to you about it. I was in a conference when you made it and I did not agree with some of the minor changes, and I thought we could see to that together. Then I followed the activity on your branch and noticed you have not worked on it anymore, so I didn't bother.

My two cents about the dev: I have recently noticed you can easily get CSV from OrientDB's REST API. The process of JSON conversion is very slow and error prone in R. Actually I have noticed several bugs in the current version of the package. If I were you, I would plan on fetching all responses in CSV rather than JSON (if the objective is having data frames in the end). That will reduce the complexity of the whole process and will allow you to move much faster. It will also make it unnecessary to optimize your code, given that you will write very little code!

**Comment by pmig** [comment]
Hi @retrography ,

Thank you for your comment.
For my part. I also currently work on my master thesis and need to do analysis within R and put the data back to orient.

I will now work on the document API #5  as soon as I have we have a new feature ready.
We should merge together all our projects!

**Comment by retrography** [comment]
Hi @pmig,

Again, I am glad you are working on it. If you guys have time feel free to take the project over. 

This used to be a fork of my repo, but it seems @mkbouaziz has reimported the repo / removed the fork, which makes it quite complicated to merge. I think it is important to keep this as a fork of my repo, given that my repo is referenced on OrientDB's website. Anyway, let's come up with a solution for that. 

I am more interested in interfacing between OrientDB and igraph. If you guys get the basics done, I will put a weekend on interfacing. It would be amazing to be able to load a whole graph from OrientDB right into igraph (or statute) for analysis!

**Comment by pmig** [comment]
Yes I definitely saw this problem and YES we have to work on it!

We also have to think about breaking the API. Right of now @retrography  and @mkbouaziz  packages already have different methods for executing a query AND accessing the database.

We should either revert these changes or create a v1 / v2 version and keep the old one unmaintained.

**Comment by retrography** [comment]
@pmig That's exactly what I wanted to discuss with @mkbouaziz when I took long to respond to his pull request!
